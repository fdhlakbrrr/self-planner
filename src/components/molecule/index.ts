export * from "./headingText";
export * from "./cardMenu";
export * from "./tooltipWrapper"
export * from "./questionTooltip"