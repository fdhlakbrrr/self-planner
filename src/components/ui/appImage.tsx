import Image from "next/image";
import MoneyPlan from "@/assets/svg/money-plan-25.svg";

interface Props {
  name: string;
  width?: number;
  height?: number;
}

export const AppImage = ({ name, width = 50, height = 50 }: Props) => {
  switch (name) {
    case "money":
      return (
        <Image alt="money plan" src={MoneyPlan} width={width} height={height} />
      );
  }
};
