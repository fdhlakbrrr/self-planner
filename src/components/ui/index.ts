export * from "./button";
export * from "./text";
export * from "./container";
export * from "./tooltip"
export * from "./appImage"
export * from "./card"
