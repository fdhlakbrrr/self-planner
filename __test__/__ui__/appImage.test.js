import { AppImage } from "../../src/components/ui";
import { render } from "@testing-library/react";

it("renders correctly", () => {
  const { container } = render(<AppImage name="money" />);

  expect(container).toMatchSnapshot();
});
