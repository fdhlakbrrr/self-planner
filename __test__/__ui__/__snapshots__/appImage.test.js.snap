// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <img
    alt="money plan"
    data-nimg="1"
    decoding="async"
    height="50"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
    style="color: transparent;"
    width="50"
  />
</div>
`;
